config/config.go

```diff
@@ -158,22 +162,90 @@ var (
 
 // Config is the top-level configuration for Prometheus's config files.
 type Config struct {
-	GlobalConfig   GlobalConfig    `yaml:"global"`
-	AlertingConfig AlertingConfig  `yaml:"alerting,omitempty"`
-	RuleFiles      []string        `yaml:"rule_files,omitempty"`
-	ScrapeConfigs  []*ScrapeConfig `yaml:"scrape_configs,omitempty"`
+	GlobalConfig      GlobalConfig    `yaml:"global"`
+	AlertingConfig    AlertingConfig  `yaml:"alerting,omitempty"`
+	RuleFiles         []string        `yaml:"rule_files,omitempty"`
+	ScrapeConfigs     []*ScrapeConfig `yaml:"scrape_configs,omitempty"`
+	ScrapeConfigFiles []string        `yaml:"scrape_config_files,omitempty"`
 
 	RemoteWriteConfigs []*RemoteWriteConfig `yaml:"remote_write,omitempty"`
 	RemoteReadConfigs  []*RemoteReadConfig  `yaml:"remote_read,omitempty"`
 }
 
+// ScrapeConfigFile is the configuration for scrape config files.
+type ScrapeConfigFile struct {
+	ScrapeConfigs []*ScrapeConfig `yaml:"scrape_configs"`
+}
+
+// LoadScrapeConfigFiles parses the given YAML files from scrape_config_files
+// and adds ScrapeConfigs from all of them.
+func (c *Config) LoadScrapeConfigFiles() error {
+	var scrapeConfigFiles []string
+	for _, pat := range c.ScrapeConfigFiles {
+		fs, err := filepath.Glob(pat)
+		if err != nil {
+			// The only error can be a bad pattern.
+			return errors.Wrapf(err, "error retrieving scrape config files for %s", pat)
+		}
+		// There is no scrape config file that matches pattern.
+		if len(fs) == 0 {
+			return errors.Errorf("no scrape config files found for %s", pat)
+		}
+		scrapeConfigFiles = append(scrapeConfigFiles, fs...)
+	}
+
+	// Job names from main config file.
+	jobNames := map[string]struct{}{}
+	for _, scfg := range c.ScrapeConfigs {
+		jobNames[scfg.JobName] = struct{}{}
+	}
+
+	for _, scrapeConfigFile := range scrapeConfigFiles {
+		scfgFile, err := LoadScrapeConfigFile(scrapeConfigFile)
+		if err != nil {
+			return err
+		}
+		for _, scfg := range scfgFile.ScrapeConfigs {
+			err := c.checkScrapeConfig(scfg)
+			if err != nil {
+				return err
+			}
+			if _, ok := jobNames[scfg.JobName]; ok {
+				return errors.Errorf("found multiple scrape configs with job name %q", scfg.JobName)
```

Comment:
It seems we should do this check just once, and not one time for the main file and one time for the secondary files.